fmc_rest_client/core/base_resources.py

```diff
@@ -214,4 +214,10 @@ def get_api_path(self):
     def json_ignore_attrs():
         return ['container']
 
+class Device(NamedResource):
+    def __init__(self, name, id=None):
+        super().__init__(name, id)
+        self.metadata = Metadata()
 
+    def get_api_path(self):
+        return self._get_api_base() + '/devices/' + self._get_resource_suffix()
```

Comment:
This will be an exception, the URL for devices is  .../devices/devicerecords/<uuid>. 
So if you want to keep the class name as Device, instead DeviceRecord which make a lot of sense, for this you need to make it as - 
 return self._get_api_base() + '/devices/devicerecords'

Reply:
Just this much - 
return self._get_api_base() + '/devices/devicerecords'

You don;t need self._get_resource_suffix(), as it will add again 'devices' at the end.